aws/codegen-aws-sdk/src/main/kotlin/software/amazon/smithy/rustsdk/RegionDecorator.kt

```diff
@@ -205,6 +205,17 @@ class RegionProviderConfig(codegenContext: ClientCodegenContext) : ConfigCustomi
                         *codegenScope,
                     )
                 }
+                is ServiceConfig.DefaultForTests -> {
+                    // this was added later, for backwards compat we only set a default if a region hasn't already been set by user
+                    rustTemplate(
+                        """
+                        if ${section.configBuilderRef}.config.load::<#{Region}>().is_none() {
+                            self.set_region(#{Some}(#{Region}::new("us-east-1")));
```

Comment:
Is it worth a BMV for this or other opt-out? I feel like there is a very high probability that _someones_ unit test will be broken by this and although the fix won't be hard they might appreciate a way to hold back to keep their tests working